### DISCUSSION TITLE
Download function gives a error (Invalid Block Size)

### DISCUSSION THREAD

**Discussion by iambeluchi** [gijzelaerr/python-snap7#267] [Q&A] [answered]
Hello Team, 

I am here again... lol

I am trying to download an OB content to the PLC, and I keep getting an invalid block size error.  

Please see my code below:

          import snap7
          import scapy
          from scapy.all import *
          from scapy.all import wrpcap
          import re
          import binascii

          plc = snap7.client.Client()
          plc.connect('192.168.0.1', 0, 2)
          db_number = 100
          start_add = 1
          packets = rdpcap(r"s7comm_downloading_block_db1.pcap")
          a = len(packets)

          for i in range(1, a):
           if (len(packets[i]) > 200):
           for pkt in packets[i]:
            ab = pkt[Raw]
            #print (ab)
            type(ab)
            pkt = binascii.hexlify(bytes(ab))
            sea_byte = pkt.find(b'32')
            data_protocol = pkt [sea_byte:sea_byte+2]
            response_msg = pkt.find (b'fb')

            data_funct_codes = """{
                                    "00":"Cpu_services",
                                    "f0":"Setup_communication",
                                    "04":"Read_Variable",
                                    "05":"Write_Variable",
                                    "1a":"Request_download",
                                    "1b":"Download_block",
                                    "1c":"Download_ended",
                                    "1d":"Start_upload",
                                    "1e":"U_Upload",
                                    "1f":"End_upload",
                                    "28":"Plc_Control",
                                    "29":"Plc_Stop"
                                    }"""

                search_download_block = pkt.find(b'1b')
               search_upload_block = pkt.find(b'1e')
               search_data_block = pkt.find(b'7070')

            if (search_download_block > 0):
                print('this is a download code')
                print("The response to this request is:")
                download_res = (pkt[response_msg + 2:])
                b = download_res
                g = bytearray(b)
                print(g)

                plc.ab_write(0, g)
                plc.download(g, 1)

             plc.plc_stop()
             plc.plc_hot_start()

             print("Close communication PLC")

             plc.disconnect()

The plc.ab_write part works. But the plc.download keeps giving error. The code is to download the data part of a pcap file to the PLC. 

Please I need help... I saw a related article but cant seem to make sense of it.

**Comment by gijzelaerr** [comment, marked as answer]
Closing this due to age and inactivity. If you're still experiencing this issue with a current version of python-snap7, please open a new discussion.
